### ISSUE TITLE
Support for tokenizers without bos token

### ISSUE THREAD

**Issue by alessiodevoto** [NVIDIA/logits-processor-zoo#24]
**Issue**
Some tokenizers do not have a bos token, or do not add it.  This should be dealt with in the encode function here: https://github.com/NVIDIA/logits-processor-zoo/blob/4e328104702020ab1e71c427277a16d3d9d2a77a/logits_processor_zoo/utils.py#L26 

This 2 should be 1 if the bos token is not present.

**Example of unexpected behavior**
This can give raise to unexpected behavior, for example: 

```
example_prompts = [
"""
Repeat the following phrase: I love Lasagna <3.
""",
]

runner = LLMRunner()
runner.generate_response(example_prompts,
                         [TriggerPhraseLogitsProcessor("Pasta", "Lasagna", runner.tokenizer, batch_size=1, 
                                                       trigger_count=2, trigger_after=False)],
                        max_tokens=10)
```

Expected behavior: raise an Error, because "Lasagna" contains two tokens

Current behavior:  
```
LLM response:
I love Lasastaasta 3.
```
